data/pki/meson.build

```diff
@@ -1,7 +1,7 @@
 # only install files that are going to be used
-if libjcat.version().version_compare ('>= 0.1.9')
-  supported_gpg = libjcat.get_pkgconfig_variable('supported_gpg')
-  supported_pkcs7 = libjcat.get_pkgconfig_variable('supported_pkcs7')
+if meson.version().version_compare('>=0.51.0')
+  supported_gpg = libjcat.get_variable('supported_gpg', default_value : '1')
+  supported_pkcs7 = libjcat.get_variable('supported_pkcs7', default_value : '1')
```

Comment:
This seems to fail though:

```
WARNING: Project targeting '>=0.50.0' but tried to use feature introduced in '0.51.0': dep.get_variable

data/pki/meson.build:3:2: ERROR: Function does not take positional arguments.
```

Reply:
Ah right, unfortunately the positional argument was introduced in 0.58

Also unfortunately, the Feature check for this is malformed and never prints...

So you will need to use

```meson
libjcat.get_variable(pkgconfig: 'supported_gpg', internal: 'supported_gpg', default_value: '1')
```

Reply:
https://github.com/mesonbuild/meson/pull/9675 to make this actually warn you...

Reply:
That seems to explode as well I'm afraid:

    The Meson build system
    Version: 0.53.2
    ...
    WARNING: Project targeting '>=0.50.0' but tried to use feature introduced in '0.51.0': dep.get_variable
    data/pki/meson.build:3: WARNING: Passed invalid keyword argument "internal".
    WARNING: This will become a hard error in the future.
      File "/usr/lib/python3/dist-packages/mesonbuild/mesonmain.py", line 129, in run
        return options.run_func(options)
      File "/usr/lib/python3/dist-packages/mesonbuild/msetup.py", line 245, in run
        app.generate()
      File "/usr/lib/python3/dist-packages/mesonbuild/msetup.py", line 159, in generate
        self._generate(env)
      File "/usr/lib/python3/dist-packages/mesonbuild/msetup.py", line 192, in _generate
        intr.run()
      File "/usr/lib/python3/dist-packages/mesonbuild/interpreter.py", line 4167, in run
        super().run()
      File "/usr/lib/python3/dist-packages/mesonbuild/interpreterbase.py", line 412, in run
        self.evaluate_codeblock(self.ast, start=1)
      File "/usr/lib/python3/dist-packages/mesonbuild/interpreterbase.py", line 436, in evaluate_codeblock
        raise e
      File "/usr/lib/python3/dist-packages/mesonbuild/interpreterbase.py", line 430, in evaluate_codeblock
        self.evaluate_statement(cur)
      File "/usr/lib/python3/dist-packages/mesonbuild/interpreterbase.py", line 451, in evaluate_statement
        return self.evaluate_if(cur)
      File "/usr/lib/python3/dist-packages/mesonbuild/interpreterbase.py", line 530, in evaluate_if
        self.evaluate_codeblock(i.block)
      File "/usr/lib/python3/dist-packages/mesonbuild/interpreterbase.py", line 436, in evaluate_codeblock
        raise e
      File "/usr/lib/python3/dist-packages/mesonbuild/interpreterbase.py", line 430, in evaluate_codeblock
        self.evaluate_statement(cur)
      File "/usr/lib/python3/dist-packages/mesonbuild/interpreterbase.py", line 441, in evaluate_statement
        return self.function_call(cur)
      File "/usr/lib/python3/dist-packages/mesonbuild/interpreterbase.py", line 788, in function_call
        return func(node, posargs, kwargs)
      File "/usr/lib/python3/dist-packages/mesonbuild/interpreterbase.py", line 285, in wrapped
        return f(*wrapped_args, **wrapped_kwargs)
      File "/usr/lib/python3/dist-packages/mesonbuild/interpreterbase.py", line 174, in wrapped
        return f(*wrapped_args, **wrapped_kwargs)
      File "/usr/lib/python3/dist-packages/mesonbuild/interpreter.py", line 3689, in func_subdir
        self.evaluate_codeblock(codeblock)
      File "/usr/lib/python3/dist-packages/mesonbuild/interpreterbase.py", line 436, in evaluate_codeblock
        raise e
      File "/usr/lib/python3/dist-packages/mesonbuild/interpreterbase.py", line 430, in evaluate_codeblock
        self.evaluate_statement(cur)
      File "/usr/lib/python3/dist-packages/mesonbuild/interpreterbase.py", line 441, in evaluate_statement
        return self.function_call(cur)
      File "/usr/lib/python3/dist-packages/mesonbuild/interpreterbase.py", line 788, in function_call
        return func(node, posargs, kwargs)
      File "/usr/lib/python3/dist-packages/mesonbuild/interpreterbase.py", line 285, in wrapped
        return f(*wrapped_args, **wrapped_kwargs)
      File "/usr/lib/python3/dist-packages/mesonbuild/interpreterbase.py", line 174, in wrapped
        return f(*wrapped_args, **wrapped_kwargs)
      File "/usr/lib/python3/dist-packages/mesonbuild/interpreter.py", line 3689, in func_subdir
        self.evaluate_codeblock(codeblock)
      File "/usr/lib/python3/dist-packages/mesonbuild/interpreterbase.py", line 436, in evaluate_codeblock
        raise e
      File "/usr/lib/python3/dist-packages/mesonbuild/interpreterbase.py", line 430, in evaluate_codeblock
        self.evaluate_statement(cur)
      File "/usr/lib/python3/dist-packages/mesonbuild/interpreterbase.py", line 451, in evaluate_statement
        return self.evaluate_if(cur)
      File "/usr/lib/python3/dist-packages/mesonbuild/interpreterbase.py", line 530, in evaluate_if
        self.evaluate_codeblock(i.block)
      File "/usr/lib/python3/dist-packages/mesonbuild/interpreterbase.py", line 436, in evaluate_codeblock
        raise e
      File "/usr/lib/python3/dist-packages/mesonbuild/interpreterbase.py", line 430, in evaluate_codeblock
        self.evaluate_statement(cur)
      File "/usr/lib/python3/dist-packages/mesonbuild/interpreterbase.py", line 443, in evaluate_statement
        return self.assignment(cur)
      File "/usr/lib/python3/dist-packages/mesonbuild/interpreterbase.py", line 1064, in assignment
        value = self.evaluate_statement(node.value)
      File "/usr/lib/python3/dist-packages/mesonbuild/interpreterbase.py", line 445, in evaluate_statement
        return self.method_call(cur)
      File "/usr/lib/python3/dist-packages/mesonbuild/interpreterbase.py", line 828, in method_call
        return obj.method_call(method_name, args, kwargs)
      File "/usr/lib/python3/dist-packages/mesonbuild/interpreterbase.py", line 325, in method_call
        return method(args, kwargs)
      File "/usr/lib/python3/dist-packages/mesonbuild/interpreterbase.py", line 231, in wrapped
        return f(*wrapped_args, **wrapped_kwargs)
      File "/usr/lib/python3/dist-packages/mesonbuild/interpreterbase.py", line 125, in wrapped
        return f(*wrapped_args, **wrapped_kwargs)
      File "/usr/lib/python3/dist-packages/mesonbuild/interpreterbase.py", line 174, in wrapped
        return f(*wrapped_args, **wrapped_kwargs)
      File "/usr/lib/python3/dist-packages/mesonbuild/interpreter.py", line 488, in variable_method
        return self.held_object.get_variable(**kwargs)
    TypeError: get_variable() got an unexpected keyword argument 'internal'
    Traceback (most recent call last):
      File "./contrib/ci/check-abi", line 144, in <module>
        new_tree = build_install(args.new)
      File "./contrib/ci/check-abi", line 83, in build_install
        subprocess.check_call(
      File "/usr/lib/python3.8/subprocess.py", line 364, in check_call
        raise CalledProcessError(retcode, cmd)
    subprocess.CalledProcessError: Command '['meson', '_build', '--prefix=/usr', '--libdir=lib', '-Db_coverage=false', '-Dgtkdoc=false', '-Ddocs=none', '-Dgusb:docs=false', '-Dtests=false']' returned non-zero exit status 2.

Reply:
I should have looked more closely at the docs. :( That was added in 0.54

Assuming that subproject builds always support it seems reasonable.